tools/bug-report/pkg/config/config.go

```diff
@@ -0,0 +1,224 @@
+//
+// 3. Pods with "prometheus" in the name, except those with
+// the annotation "internal=true":
+//      include:
+//        */*/*prometheus*
+//      exclude:
+//        */*/*prometheus*/*/internal=true
+//
+// 4. Container logs for all containers called "istio-proxy":
+//      include:
+//        */*/*/*/*/istio-proxy
+//
+type SelectionSpec struct {
+	Namespaces  []string
+	Deployments []string
+	Pods        []string
+	Containers  []string
+	Labels      map[string]string
+	Annotations map[string]string
+}
+
+// BugReportConfig controls what is captured and Include in the kube-capture tool
+// archive.
+type BugReportConfig struct {
+	// KubeConfigPath is the path to kube config file.
+	KubeConfigPath string `json:"kubeConfigPath,omitempty"`
+	// Context is the cluster Context in the kube config
+	Context string `json:"context,omitempty"`
+
+	// IstioNamespace is the namespace where the istio control plane is installed.
+	IstioNamespace string `json:"istioNamespace,omitempty"`
+
+	// DryRun controls whether logs are actually captured and saved.
+	DryRun bool `json:"dryRun,omitempty"`
+
+	// FullSecrets controls whether secret contents are included.
+	FullSecrets bool `json:"fullSecrets,omitempty"`
+
+	// CommandTimeout is the maximum amount of time running the command
+	// before giving up, even if not all logs are captured. Upon timeout,
+	// the command creates an archive with only the logs captured so far.
+	CommandTimeout Duration `json:"commandTimeout,omitempty"`
+	// MaxArchiveSizeMb is the maximum size of the archive in Mb. When
+	// the maximum size is reached, logs are selectively discarded based
+	// on importance metrics.
+	MaxArchiveSizeMb int32 `json:"maxArchiveSizeMb,omitempty"`
+
+	// Include is a list of SelectionSpec entries for resources to include.
+	Include []*SelectionSpec `json:"include,omitempty"`
+	// Exclude is a list of SelectionSpec entries for resources t0 exclude.
+	Exclude []*SelectionSpec `json:"exclude,omitempty"`
+
+	// StartTime is the start time the the log capture time range.
+	// If set, Since must be unset.
+	StartTime time.Time `json:"startTime,omitempty"`
+	// EndTime is the end time the the log capture time range.
+	// Default is now.
+	EndTime time.Time `json:"endTime,omitempty"`
+	// Since defines the start time the the log capture time range.
+	// StartTime is set to EndTime - Since.
+	// If set, StartTime must be unset.
+	Since Duration `json:"since,omitempty"`
+
+	// CriticalErrors is a list of glob pattern matches for errors that,
+	// if found in a log, set the highest priority for the log to ensure
+	// that it is Include in the capture archive.
+	CriticalErrors []string `json:"criticalErrors,omitempty"`
+	// IgnoredErrors are glob error patterns which are ignored when
+	// calculating the error heuristic for a log.
+	IgnoredErrors []string `json:"ignoredErrors,omitempty"`
+
+	// GCSURL is an URL to the GCS bucket where the archive is uploaded.
+	GCSURL string `json:"gcsURL,omitempty"`
```

Comment:
Is this being used?  I couldn't see it being used.  I suggest the upload be separate from istioctl anyway.

Reply:
regarding tempdir (can't reply to that above for some reason) it's not possible in this case because mkdir is done just before the file is written (the dir hierarchy is not known at startup). 
the upload was considered highly desired in the review doc - stewart has implemented it and will integrate after this merges.